bindata/network/openshift-sdn/sdn.yaml

```diff
@@ -66,7 +66,56 @@ spec:
         - |
           #!/bin/bash
           set -euo pipefail
+{{- if .IsNetworkTypeMigration }}
+          set +e
 
+          SDN_SUBNET=$(oc get hostsubnet ${K8S_NODE_NAME} -o jsonpath="{.subnet}")
+          echo "sdn hostsubnet:" ${SDN_SUBNET}
+          OVN_SUBNET=$(oc get node ${K8S_NODE_NAME} -o jsonpath='{.metadata.annotations.k8s\.ovn\.org/node-subnets}'| grep -o  -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\/[0-9]{1,3}')
+          echo "ovnkube node-subnets annotation:" ${OVN_SUBNET}
+          set -ex
+
+          if [ ! -z "${SDN_SUBNET}" ] && [ ! -z "${OVN_SUBNET}" ] && [ "${SDN_SUBNET}" != "${OVN_SUBNET}" ]; then
+              echo "hostsubnet mismatch"
+              exit 1 
+          fi
+
+          if [ ! -z "${SDN_SUBNET}" ]; then
+              echo "sdn hostsubnet is set"
+              SUBNET=${SDN_SUBNET}
+          else
+              echo "sdn hostsubnet is unset"
+              if [ ! -z "${OVN_SUBNET}" ]; then
+                  echo "hostsubnet CR is not found, use the value from ovnkube annotation"
+                  SUBNET=${OVN_SUBNET}
+                  NODE_UID=$(oc get node ${K8S_NODE_NAME} -o jsonpath='{.metadata.uid}')
+                  NODE_IP=$(oc get nodes ${K8S_NODE_NAME} -o jsonpath='{.status.addresses[?(@.type=="InternalIP")].address}')
+                  echo "create hostsubnet CR for this node"
+          cat <<EOF | oc apply -f -
+          apiVersion: network.openshift.io/v1
+          kind: HostSubnet
+          metadata:
+            name: ${K8S_NODE_NAME}
+            annotations:
+              pod.network.openshift.io/node-uid: ${NODE_UID}
+          host: ${K8S_NODE_NAME}
+          hostIP: ${NODE_IP}
+          subnet: ${SUBNET}
+          EOF
+              else
+                  echo "cannot discover host subnet"
+                  exit 1
+              fi
+          fi
+          if ovs-vsctl br-exists br-ex; then
+            echo "br-ex exists, sleep..."
+            sleep infinity
```

Comment:
To avoid delays in killing the container we could use the following instead:
```
trap : TERM INT; sleep infinity & wait
```

Additionally would it be good to put an exit after the sleep? not sure about the conventions/best practice here.